plugins/modules/smart_proxy.py

```diff
@@ -101,13 +110,50 @@ def main():
         foreman_spec=dict(
             name=dict(required=True),
             url=dict(required=True),
+            lifecycle_environments=dict(required=False, type='entity_list'),
             download_policy=dict(required=False, choices=['background', 'immediate', 'on_demand']),
         ),
-        required_plugins=[('katello', ['download_policy'])],
+        required_plugins=[('katello', ['lifecycle_environments', 'download_policy'])],
     )
 
     with module.api_connection():
-        module.run()
+        handle_lifecycle_environments = not module.desired_absent and 'lifecycle_environments' in module.foreman_params
+        if handle_lifecycle_environments:
+            module.lookup_entity('lifecycle_environments')
+
+        lifecycle_environments = module.foreman_params.pop('lifecycle_environments', None)
+        smart_proxy = module.run()
+
+        payload = {
+            'id': smart_proxy['id'],
+        }
+        smart_proxy_content = module.resource_action('capsule_content', 'lifecycle_environments', payload, record_change=False)
```

Comment:
```suggestion
        current_lces = module.resource_action('capsule_content', 'lifecycle_environments', payload, record_change=False)
```

this would make the lines short enough and is IMHO more correct, as the action returns you a list of LCEs, not a smart proxy with content object.

Reply:
Yeah, that sounds better, I now used the name of the endpoint, sec, disregard the current push :-)